NXDOC-2216: Update package manifest doc to reflect target platform ve… (#1593)
nuxeo/doc.nuxeo.com-content: src/nxdoc/nuxeo-add-ons/creating-nuxeo-packages/package-manifest.md

  <description>
    <p>Put here an HTML description of your Nuxeo Package.</p>
  </description>
  <target-platform>
    <name>server</name>
    <version>[11.10,12)</version>
  </target-platform>
</package>

```

This is a minimal package manifest. It is defining a package nuxeo-package at version 1.0.0 and of type addon.

> The package can be installed on platforms form server-11.10 (included) to server-12.0 (exculded).

Reviewer: ```suggestion
The package can be installed on platforms from server-11.10 (included) to server-12.0 (excluded).
```

  See [org.nuxeo.connect.update.NuxeoValidationState.](https://qa.nuxeo.org/jenkins/job/master/job/nuxeo-connect-master/site/nuxeo-connect-client/apidocs/org/nuxeo/connect/update/NuxeoValidationState.html)
- `production-state`: A production status: `proto`, `testing`, `production_ready`.</br>
  See [org.nuxeo.connect.update.ProductionState](https://qa.nuxeo.org/jenkins/job/master/job/nuxeo-connect-master/site/nuxeo-connect-client/apidocs/org/nuxeo/connect/update/ProductionState.html).
- `license`: Open field: GPL, BSD, LGPL...
- `license-url`: If no URL is provided, then a `license.txt` file should be included in the package.
- `visibility`: `PRIVATE` (restricted to specific users and/or projects), `MARKETPLACE` (restricted to registered users), `DEV` (restricted to registered users) or `PUBLIC` (no restriction). The visibility determines the channel where the package will be distributed and how it can be installed.</br>
  See [org.nuxeo.connect.update.PackageVisibility](https://qa.nuxeo.org/jenkins/job/master/job/nuxeo-connect-master/site/nuxeo-connect-client/apidocs/org/nuxeo/connect/update/PackageVisibility.html).

> - `platforms`: The list of platforms supported by this package.

Reviewer: `platforms` hasn't been removed

Maintenance branches are usually named `major.minor` or `major.minor_LTS`: for instance, `1.0`, `1.1` and `1.2` or `1.0_8.10`, `1.1_9.10` and `1.2_10.10`.

## Target platform

The prefered way to define on which Nuxeo platform a package can be installed is the following:

```xml
  <target-platform>
    <name>[compatible distribution name]</name>
    <version>[range of compatible distribution versions]</version>
  </target-platform>

```

> Where the compatible distribution name must refers to the `org.nuxeo.distribution.name` property of the Nuxeo platform and the range of compatible distribution versions is defined using a *maven-like* version range syntax and refers to the `org.nuxeo.distribution.version` property of the Nuxeo platform.

Reviewer: ```suggestion
The compatible distribution name must refer to the `org.nuxeo.distribution.name` property of the Nuxeo platform. 

The range of compatible distribution versions is defined using a *maven-like* version range syntax and refers to the `org.nuxeo.distribution.version` property of the Nuxeo platform.
```

### Constraints

Here are the constraints to be followed when defining the `<target-platform>` in your package.xml:

- Both `<name>` and `<version>` tags are **required** inside `<target-platform>`
- The `<name>` is case sensitive and must be an **exact match** of the `org.nuxeo.distribution.name` property of the Nuxeo platform
- The `<version>` syntax follows these rules

    | Version Range	    | Meaning         |
    | --------- | --------------- |
    | 1.0	    | x == 1.0        |
    | (,1.0]    | x <= 1.0        |
    | (,1.0)    | x < 1.0         |
    | [1.0]	    | x == 1.0        |
    | [1.0,)    | x >= 1.0        |
    | (1.0,)    | x > 1.0         |
    | (1.0,2.0)	| 1.0 < x < 2.0   |
    | [1.0,2.0]	| 1.0 <= x <= 2.0 |

> - The `<version>` must refers to the `org.nuxeo.distribution.version` property of the Nuxeo platform

Reviewer: ```suggestion
- The `<version>` must refer to the `org.nuxeo.distribution.version` property of the Nuxeo platform
```

> - As only one range can be defined, **exclusion is not possible**. To achieve the same purpose, you should use the `<conflicts>` tag and/or do separate specific releases of your package

Reviewer: ```suggestion
- Only one range can be defined and **exclusion is not possible**. To achieve the same purpose, use the `<conflicts>` tag and/or do separate specific releases of your package
```

### Patterns (former way)

Although it is not the recommanded way, it is still possible to use the former `<platforms>` tag to declare pattern(s) matching the *[name]-[version]* couple of the targeted Nuxeo platforms:

```xml
  <platforms>
    <platform>server-11.10</platform>
    <platform>server-11.10-HF*</platform>
  </platforms>

```
{{#> callout type='note' }}

> The new `<target-platform>` tag take precedence on the former `<platforms>` tag if both are defined in the package.xml

Reviewer: ```suggestion
The new `<target-platform>` tag takes precedence on the former `<platforms>` tag if both are defined in the package.xml
```
